src/analysis/buildTypeWeights.ts

```diff
@@ -193,18 +220,95 @@ function parseTypes(schema: GraphQLSchema, typeWeights: TypeWeightSet): TypeWeig
                     weight: typeWeights.scalar,
                 };
             } else if (isUnionType(currentType)) {
-                // FIXME: will need information on fields inorder calculate comlpextiy
-                result[typeName] = {
-                    fields: {},
-                    weight: typeWeights.object,
-                };
+                unions.push(currentType);
             } else {
+                // FIXME: Scalar types are listed here throw new Error(`ERROR: buildTypeWeight: Unsupported type: ${currentType}`);
```

Comment:
Is it only Scalar Types trowing an error here? Could you add a check to throw an error only if it's not a scalar type?

Reply:
It appears so, I've added this check. We may want to silence the error later or at least wrap this function in a try catch when being used in the middleware